src/channel_manager.ts

```diff
@@ -0,0 +1,517 @@
+
+    const channelType = event.channel_type;
+    const channelId = event.channel_id;
+
+    if (channelType && channelId) {
+      const targetChannel = this.client.channel(channelType, channelId);
+      const targetChannelIndex = channels.indexOf(targetChannel);
+      const targetChannelExistsWithinList = targetChannelIndex >= 0;
+
+      const isTargetChannelPinned = isChannelPinned(targetChannel);
+      const isTargetChannelArchived = isChannelArchived(targetChannel);
+
+      const considerArchivedChannels = shouldConsiderArchivedChannels(filters);
+      const considerPinnedChannels = shouldConsiderPinnedChannels(sort);
+
+      if (
+        // filter is defined, target channel is archived and filter option is set to false
+        (considerArchivedChannels && isTargetChannelArchived && !filters.archived) ||
+        // filter is defined, target channel isn't archived and filter option is set to true
+        (considerArchivedChannels && !isTargetChannelArchived && filters.archived) ||
+        // sort option is defined, target channel is pinned
+        (considerPinnedChannels && isTargetChannelPinned) ||
+        // list order is locked
+        this.options.lockChannelOrder ||
+        // target channel is not within the loaded list and loading from cache is disallowed
+        (!targetChannelExistsWithinList && !this.options.allowNewMessagesFromUnfilteredChannels)
+      ) {
+        return;
+      }
+
+      this.setChannels(
+        moveChannelUpwards({
+          channels,
+          channelToMove: targetChannel,
+          channelToMoveIndexWithinChannels: targetChannelIndex,
+          sort,
+        }),
+      );
+    }
+  };
+
+  private notificationNewMessageHandler = async (event: Event<SCG>) => {
+    const { id, type } = event?.channel ?? {};
+
+    if (id && type) {
+      const channel = await getAndWatchChannel({
+        client: this.client,
+        id,
+        type,
+      });
+      const { channels, pagination } = this.state.getLatestValue();
+      const { filters, sort } = pagination ?? {};
+
+      const considerArchivedChannels = shouldConsiderArchivedChannels(filters);
+      if (isChannelArchived(channel) && considerArchivedChannels && !filters.archived) {
+        return;
+      }
+
+      if (!this.options.allowNewMessagesFromUnfilteredChannels) {
+        return;
+      }
+
+      this.setChannels(
+        moveChannelUpwards({
+          channels,
+          channelToMove: channel,
+          channelToMoveIndexWithinChannels: -1,
+          sort,
+        }),
+      );
+    }
+  };
+
+  private channelVisibleHandler = async (event: Event<SCG>) => {
+    const { channels, pagination } = this.state.getLatestValue();
+    const { sort } = pagination ?? {};
+    if (channels && event.channel_type && event.channel_id) {
+      const channel = await getAndWatchChannel({
+        client: this.client,
+        id: event.channel_id,
+        type: event.channel_type,
+      });
+
+      this.setChannels(
```

Comment:
The fix for this is already [ported to develop](https://github.com/GetStream/stream-chat-react-native/pull/2925/files).